### PR TITLE
tests: Add macros to assert that a copied string has a desired value

### PR DESCRIPTION
**Pull request by smcv**
* tests: Add macros to assert that a copied string has a desired value
    
    Suggested by Patrick during review of #5691.

* testlibrary: Make use of assert_cmpstr_free_lhs(), etc.

### CODE REVIEW

**Comment by smcv** [comment]
Not yet tested (with AddressSanitizer or otherwise) but I wanted to get a version up so @TingPing could opine. What do you think about this as a reasonably non-ugly implementation?

**Comment by smcv** [comment]
In principle we could also have a `assert_cmpstr_free_rhs` (but there would be no uses yet, so I didn't add it) and/or a `assert_cmpstr_free_both` (but there would be only one use of it, so it didn't seem worth it).

File: tests/testlib.h [hunk 1]
```diff
@@ -46,4 +46,11 @@ G_DEFINE_AUTOPTR_CLEANUP_FUNC (TestsStdoutToStderr, tests_stdout_to_stderr_end);
 #define TESTS_SCOPED_STDOUT_TO_STDERR \
   G_GNUC_UNUSED g_autoptr(TestsStdoutToStderr) _tests_stdout_to_stderr = tests_stdout_to_stderr_begin ()
 
+/* Implemented as a macro so we get useful line numbers in assertion failures */
+#define assert_cmpstr_free_lhs(owned_string, op, unowned_string) \
```
Comment (TingPing) [inline tests/testlib.h:50]: `op` unused?

**Review by TingPing — APPROVED** [review]
Exactly what I was imagining, much nicer :)

File: tests/testlib.h (hunk 1 above)
Comment (smcv) [reply tests/testlib.h]: Oops, yes: it was blindly assuming `==`. Now fixed.

**Comment by smcv** [comment]
Updated to have three macros, here's a summary of using them correctly:

```
assert_cmpstr_free_lhs (g_strdup ("hello"), ==, "hello");
assert_cmpstr_free_rhs ("hello", ==, g_strdup ("hello"));
assert_cmpstr_free_both (g_strdup ("hello"), ==, g_strdup ("hello"));
```

(obviously in real life you'd be using something more like `flatpak_foo_get_bar()` that returns a copied string)

**Review by TingPing — APPROVED** [review]

**Pull request merged**